sdk/core/azure-core/inc/azure/core/nullable.hpp

```diff
@@ -102,8 +102,8 @@ template <class T> class Nullable final {
   {
     if (m_hasValue)
     {
-      m_value.~T();
       m_hasValue = false;
+      m_value.~T();
```

Comment:
It would be helpful for future maintainers if we add a source comment here to explain why the line order is intentionally this way since it matters, and to not reverse it.